src/GraphQL.Serialization.Newtonsoft/Http/DocumentWriter.cs

```diff
@@ -0,0 +1,49 @@
+using System.Buffers;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace GraphQL.Http
+{
+    public class DocumentWriter : IDocumentWriter
+    {
+        private readonly JsonArrayPool _jsonArrayPool = new JsonArrayPool(ArrayPool<char>.Shared);
+        private readonly JsonSerializer _serializer;
+        internal static readonly Encoding Utf8Encoding = new UTF8Encoding(false);
+
+        public DocumentWriter()
+            : this(indent: false)
+        {
+        }
+
+        public DocumentWriter(bool indent)
+            : this(
+                indent ? Formatting.Indented : Formatting.None,
+                new JsonSerializerSettings())
+        {
+        }
+
+        public DocumentWriter(Formatting formatting, JsonSerializerSettings settings)
```

Comment:
Also `Formatting formatting` parameter is not needed, just set `JsonSerializerSettings.Formatting` property in the second ctor.

Reply:
The formatting parameter was there before, its used when the indent option is passed in, and could be used by other users.  If we modify the settings passed in, it could inadvertently affect users of the API if their settings are shared in other places of their application.

Reply:
Yes, that’s it. But this parameter is redundant. It is already contained in the second parameter. It only confuses.

> it could inadvertently affect users of the API 

Remember that you are already making backward incompatible changes. It makes no sense to try to maintain backward compatibility at a single point when everything around is already redone. Better to fix the old design flaw.

Reply:
Its removed